### PR TITLE
Tutorial

### PR DESCRIPTION
**Pull request by bencbernstein**

### CODE REVIEW

File: Apiary/ViewControllers/HiveViewController.swift [hunk 1]
```diff
@@ -32,6 +32,12 @@ final class HiveViewController: UIViewController {
     spinner.center(inView: view)
     spinner.startAnimating()
     
+    // Tutorial if the user hasn't completed it
+    guard let user = store.user else { return }
```
Comment (OLI9292) [inline Apiary/ViewControllers/HiveViewController.swift]: i don't think we need `tutorialCompleted` on the User obj.  Just forward them to the tutorial if the user doesn't exist then create them after they've completed it

File: Apiary/ViewControllers/HiveViewController.swift [hunk 2]
```diff
@@ -32,6 +32,12 @@ final class HiveViewController: UIViewController {
...
+    if !user.tutorialCompleted {
+      self.navigationController?.pushViewController(TutorialViewController(), animated: true);
```
Comment (OLI9292) [inline Apiary/ViewControllers/HiveViewController.swift]: this navigation should take place in NavigationManager

File: Apiary/ViewControllers/TutorialViewController.swift [hunk 3]
```diff
@@ -0,0 +1,105 @@
+//
+//  GameViewController.swift
+//
+import RxCocoa
+import RxGesture
+import RxRealm
+import RxSwift
+import RxSwiftExt
```
Comment (OLI9292) [inline Apiary/ViewControllers/TutorialViewController.swift]: it doesn't look like you're using some of these modules

File: Apiary/ViewControllers/TutorialViewController.swift [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+  }
+  
+  fileprivate func addNewView(view: UIView) {
+    self.view.viewWithTag(1)?.removeFromSuperview()
+    view.alpha = 0
+    UIView.animate(withDuration: 0.2, animations: { 
+      view.alpha = 1
+    })
+    
+    if let currentView = currentView as? IntroView {
```
Comment (OLI9292) [inline Apiary/ViewControllers/TutorialViewController.swift]: can you not set `currentView.delegate = self` without the if / lets?

File: Apiary/Views/TutorialViews/TutorialQuestionFour.swift [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+      $0.setImage(#imageLiteral(resourceName: "Play").withRenderingMode(.alwaysTemplate), for: .normal)
+      $0.tintColor = Palette.green.color
+      $0.titleLabel?.numberOfLines = 1
+      $0.titleLabel?.textAlignment = .center
+      $0.addTarget(self, action: #selector(self.nextQuestion), for: .touchUpInside)
+      self.addSubview($0)
+    }
+  }
+  
+  private func animate(string: String, count: Int = 0, label: UILabel) {
```
Comment (OLI9292) [inline Apiary/Views/TutorialViews/TutorialQuestionFour.swift]: it seems like if this function is being reused you would want to move it to `helpers` - that way if we change it we don't have to change it in multiple places

File: Apiary/Views/TutorialViews/TutorialQuestionOne.swift [hunk 6]
```diff
@@ -0,0 +1,152 @@
+//
+//  TutorialQuestionOne.swift
+//
+
```
Comment (OLI9292) [inline Apiary/Views/TutorialViews/TutorialQuestionOne.swift]: extra lines

File: Apiary/Views/TutorialViews/TutorialQuestionOne.swift [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+        })
+      })
+    } else {
+      UIView.animate(
+        withDuration: 0.2,
+        delay: 0.0,
+        options: [.autoreverse],
+        animations: {
+          sender.tintColor = Palette.red.color
+      }, completion: { (true) in
```
Comment (OLI9292) [inline Apiary/Views/TutorialViews/TutorialQuestionOne.swift]: `_ in` is more standard

File: Apiary/ViewControllers/TutorialViewController.swift [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+    completion.asObservable()
+      .subscribe(onNext: { completionPerc in
+        self.tutorialProgressBar.animateIncrease(completion: completionPerc)
+      })
+      .addDisposableTo(disposeBag)
+
+    addNewView(view: currentView)
+   }
+  
+  fileprivate func nextView() {
```
Comment (OLI9292) [inline Apiary/ViewControllers/TutorialViewController.swift]: i think there's a method with protocol extensions that would make this VC much simpler but if it works it's might be worth skipping

**Comment by OLI9292** [comment]
squashing those commit would be nice

File: Apiary/ViewControllers/TutorialViewController.swift (hunk 4 above)
Comment (bencbernstein) [inline Apiary/ViewControllers/TutorialViewController.swift]: not unless they're cast a view with a delegate on them. only way around this i can think of is creating a tutorialview and then subclassing again. seems like overkill right now

File: Apiary/ViewControllers/TutorialViewController.swift (hunk 8 above)
Comment (bencbernstein) [inline Apiary/ViewControllers/TutorialViewController.swift]: feel free to show me

**Pull request closed without merging**